butterfly-slack/src/main/java/com/paypal/butterfly/slack/SlackNotifierConfig.java

```diff
@@ -0,0 +1,140 @@
+package com.paypal.butterfly.slack;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.paypal.butterfly.api.TransformationListener;
+import com.paypal.butterfly.api.TransformationRequest;
+import com.paypal.butterfly.api.TransformationResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+import javax.annotation.PostConstruct;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.io.File;
+
+/**
+ * Observer that receives data from Pupa and sends it to
+ * a slack user channel when a transformation is complete.
+ *
+ * @author edhwang
+ */
+@Configuration
+public class SlackNotifierConfig {
+    private static final Logger logger = LoggerFactory.getLogger(SlackNotifierConfig.class);
+    @Bean
+    @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+    public TransformationListener slackNotification(){
+        return new TransformationListener() {
+
+            private String token;
+            private WebTarget target;
+            private String postSuccessJson;
+            private String postAbortJson;
+            private String[] butterflyUserIDs;
+
+            @PostConstruct
+            public void postConstruct() {
+                try {
+                    String postSuccessPath = new File(this.getClass().getResource("/postSuccess.json").toURI()).getAbsolutePath();
+                    String postAbortPath = new File(this.getClass().getResource("/postAbort.json").toURI()).getAbsolutePath();
+                    Client client = ClientBuilder.newClient();
+                    target = client.target("https://www.slack.com/api/");
+                    token = System.getProperty("slack.token");
+                    postSuccessJson = new String(Files.readAllBytes(Paths.get(postSuccessPath)));
+                    postAbortJson = new String(Files.readAllBytes(Paths.get(postAbortPath)));
+                    //TODO get butterflyUserID, details link and github url from Pupa
+                    butterflyUserIDs = new String[]{"edwardhwang@berkeley.edu", "ehwang0824@gmail.com"};
```

Comment:
Let's rename this to `slackUserIds`

Reply:
I changed to slackUserEmails because this should be a list of emails associated with their slack accounts